File: 단계별로_풀어보기/6. 심화 1단계/8_25206.js
// 입력
const fs = require('fs')
const inputArr = fs.readFileSync(0, 'utf-8').toString().trim().split('\n')

let sumMother = 0
let sumSon = 0

const scoreMap = {
  'A+': 4.5,
  A0: 4.0,
  'B+': 3.5,
  B0: 3.0,
  'C+': 2.5,
  C0: 2.0,
  'D+': 1.5,
  D0: 1.0,
  F: 0
}

inputArr.forEach(input => {
  const [title, mother, grade] = input.trim().split(' ')
  const motherNum = parseFloat(mother)

  if (grade !== 'P') {
    sumMother += motherNum
    sumSon += motherNum * scoreMap[grade]
  }
})

process.stdout.write(`${sumSon / sumMother}`)

File: 1. 단계별로_풀어보기/6. 심화 1단계/8_25206.py
import sys

input = sys.stdin.readline
output = sys.stdout.write

inputArr = list(list(input().strip().split(' ')) for i in range(20))

sumMother = 0
sumSon = 0

scoreMap = {
  'A+': 4.5,
  'A0': 4.0,
  'B+': 3.5,
  'B0': 3.0,
  'C+': 2.5,
  'C0': 2.0,
  'D+': 1.5,
  'D0': 1.0,
  'F': 0
}

for input in inputArr:
  title, mother, son = input

  if son != 'P':
    sumMother += float(mother)
    sumSon += float(mother) * scoreMap[son]

output(f"{sumSon / sumMother}")